test_ros_timer.test_timer_uses_sim_time fails if repeated

## Bug report

**Required Info:**

- Operating System:
  - All platforms
- Installation type:
  - source
- Version or commit hash:
  - b348a81870588ad27b6bc9434b2ec4ce861ded0f
- DDS implementation:
  - N/A
- Client library (if applicable):
  - N/A

#### Steps to reproduce issue

1. Build launch_ros
2. Run `test_ros_timer` with `--repeat-until-fail`, e.g.

```
colcon test --packages-select test_launch_ros --pytest-args -k test_ros_timer --event-handlers console_direct+ --retest-until-fail 2
```

#### Expected behavior

Test passes

#### Actual behavior

Test fails and we see one or more runtime errors that look like:

```
________________________ test_timer_uses_sim_time[2-11] ________________________
test/test_launch_ros/actions/test_ros_timer.py:154: in test_timer_uses_sim_time
    rclpy.init()                                          
/opt/ros/rolling/lib/python3.8/site-packages/rclpy/__init__.py:76: in init
    return context.init(args, domain_id=domain_id)
/opt/ros/rolling/lib/python3.8/site-packages/rclpy/context.py:70: in init
    raise RuntimeError
E   RuntimeError
```

#### Additional information

Example test failures:
- https://ci.ros2.org/view/nightly/job/nightly_linux_repeated/2411/testReport/test_launch_ros.test.test_launch_ros.actions/test_ros_timer/test_timer_uses_sim_time_2_3_/
- https://ci.ros2.org/view/nightly/job/nightly_linux_repeated/2411/testReport/test_launch_ros.test.test_launch_ros.actions/test_ros_timer/test_timer_uses_sim_time_3_3_/

--- a/test_launch_ros/test/test_launch_ros/actions/test_ros_timer.py
+++ b/test_launch_ros/test/test_launch_ros/actions/test_ros_timer.py
@@ -25,6 +25,7 @@ import launch.event_handlers
 from launch.substitutions import LaunchConfiguration
 from launch_ros.actions import RosTimer
 from launch_ros.actions import SetUseSimTime
+import pytest
 import rclpy
 from rclpy.clock import Clock, ClockType
 from rosgraph_msgs.msg import Clock as ClockMsg
@@ -148,12 +149,18 @@ def test_shutdown_preempts_timers():
     assert shutdown_reasons[0].reason == 'fast shutdown'
 
 
-def test_timer_uses_sim_time():
-    """Test that timer uses time from /clock topic."""
-    # Create clock publisher node
+@pytest.fixture
+def rclpy_node():
     rclpy.init()
     node = rclpy.create_node('test_ros_timer_action_node')
-    publisher = node.create_publisher(ClockMsg, '/clock', 10)
+    yield node
+    rclpy.shutdown()
+
+
+def test_timer_uses_sim_time(rclpy_node):
+    """Test that timer uses time from /clock topic."""
+    # Create clock publisher node
+    publisher = rclpy_node.create_publisher(ClockMsg, '/clock', 10)
 
     # Increment sim time by 100 every time callback is called
     def timer_callback(publisher, time_msg):
@@ -163,9 +170,9 @@ def test_timer_uses_sim_time():
     # For every second of system time, publish new sim time value
     callback_clock = Clock(clock_type=ClockType.SYSTEM_TIME)
     time_msg = Time(sec=0, nanosec=0)
-    node.create_timer(1, partial(timer_callback, publisher, time_msg), clock=callback_clock)
+    rclpy_node.create_timer(1, partial(timer_callback, publisher, time_msg), clock=callback_clock)
 
-    thread = threading.Thread(target=rclpy.spin, args=(node, ), daemon=True)
+    thread = threading.Thread(target=rclpy.spin, args=(rclpy_node, ), daemon=True)
     thread.start()
 
     ld = launch.LaunchDescription([

